src/Compilers/Core/Portable/SourceGeneration/Nodes/TransformNode.cs

```diff
@@ -44,23 +56,41 @@ public NodeStateTable<TOutput> UpdateStateTable(DriverStateTable.Builder builder
             // - Added: perform transform and add
             // - Modified: perform transform and do element wise comparison with previous results
 
-            var newTable = previousTable.ToBuilder();
+            var newTable = previousTable.ToBuilder(builder.DriverState.TrackIncrementalSteps);
```

Comment:
Consider making builder have a method like `CreateTableBuilder()` that returns a builder for the given table with the correct step tracking enabled/disabled, rather than exposing the property:

`var newTable = builder.CreateTableBuilder(previousTable)`